src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostTestBase.cs

```diff
@@ -9,25 +9,33 @@
 using Microsoft.AspNetCore.Razor.Test.Common;
 using Microsoft.AspNetCore.Razor.Test.Common.Workspaces;
 using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.Razor.Remote;
 using Microsoft.CodeAnalysis.Remote.Razor;
 using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Composition;
 using Xunit.Abstractions;
 
 namespace Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
 
 public abstract class CohostTestBase(ITestOutputHelper testOutputHelper) : WorkspaceTestBase(testOutputHelper)
 {
-    private IRemoteServiceProvider? _remoteServiceProvider;
+    private TestRemoteServiceInvoker? _remoteServiceInvoker;
 
-    private protected IRemoteServiceProvider RemoteServiceProvider => _remoteServiceProvider.AssumeNotNull();
+    private protected TestRemoteServiceInvoker RemoteServiceInvoker => _remoteServiceInvoker.AssumeNotNull();
 
     protected override async Task InitializeAsync()
     {
         await base.InitializeAsync();
 
-        var exportProvider = AddDisposable(await RemoteMefComposition.CreateExportProviderAsync());
-        _remoteServiceProvider = AddDisposable(new TestRemoteServiceProvider(exportProvider));
+        // Create a new isolated MEF composition but use the cached configuration for performance.
+        var configuration = await RemoteMefComposition.GetConfigurationAsync(DisposalToken);
+        var runtimeComposition = RuntimeComposition.CreateRuntimeComposition(configuration);
+        var exportProviderFactory = runtimeComposition.CreateExportProviderFactory();
+        var exportProvider = exportProviderFactory.CreateExportProvider();
```

Comment:
Consider having these 3 lines in an internal method on `RemoteMefComposition`, used by this code, and that class.